dependency.lic

```diff
@@ -772,257 +773,467 @@ class SlackbotManager
 end
 
 class SetupFiles
-  def initialize(debug)
+  # http://www.druby.org/sidruby/5-3-thread-safe-communication-using-locking-mutex-and-monitormixin.html#sec.monitormixin
+  include MonitorMixin
+
+  class FileInfo
+    attr_reader :path, :name, :mtime
+
+    def initialize(path:, name:, data:, mtime:)
+      @path = path
+      @name = name
+      @data = data
+      @mtime = mtime
+    end
+
+    def data
+      # Create deep clone of data to mitigate scripts changing settings in-memory
+      # because that leads to hard to track down bugs and unpredictable script behaviors.
+      Marshal.load(Marshal.dump(@data))
+    end
+
+    def peek(property)
+      # An efficient and safe way to return a specific property
+      # from the yaml file without the risks of passing around shallow copies.
+      # This marshals the one property which is magnitudes faster than
+      # marshalling the entire file to access one property.
+      Marshal.load(Marshal.dump(@data[property.to_sym]))
+    end
+
+    def to_s
+      File.join(@path, @name)
+    end
+
+    def inspect
+      "#<SetupFiles::FileInfo @name=#{@name}, @path=#{@path}, @mtime=#{@mtime}>"
+    end
+  end
+
+  def initialize(debug = false)
+    super() # Must call super() to initialize the MonitorMixin
+    @files_cache = {} # filename => FileInfo
```

Comment:
What does this comment mean?

Reply:
it's a map of filenames to FileInfo objects.

Reply:
> it's a map of filenames to FileInfo objects.

This, yep!

```suggestion
    @files_cache = {} # map of filenames => FileInfo objects
```